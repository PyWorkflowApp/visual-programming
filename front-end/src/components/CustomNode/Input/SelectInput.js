import React, { useState, useEffect } from 'react';
import { Form } from 'react-bootstrap';

export default function SelectInput(props) {

    const [value, setValue] = useState(props.value);
    const handleChange = (event) => {
        setValue(event.target.value);
    };

    const {keyName, onChange} = props;
    // whenever value changes, fire callback to update config form
    useEffect(() => {
            onChange(keyName, value);
        },
        [value, keyName, onChange]);

    return  (
        <Form.Control as="select" name={props.keyName}
                    value={value}
                    onChange={handleChange}>
            {props.options.map(opt =>
                <option key={opt} value={opt}>{opt}</option>
            )}
        </Form.Control>
    )
}
