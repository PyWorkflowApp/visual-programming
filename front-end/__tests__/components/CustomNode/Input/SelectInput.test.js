import React from 'react'
import { shallow, mount } from 'enzyme';
import SelectInput from '../../../../src/components/CustomNode/Input/SelectInput';

describe('Validates SelectInput', () => {
  it('Display SelectInput', () => {
    const options =["area", "bar", "line", "point"];
    const input = shallow(<SelectInput
          options={options}
          />);
    expect(input).toMatchSnapshot();
  });

  it('SelectInput onChange', () => {
    const onChange = jest.fn(() => []);
    const options =["area", "bar", "line", "point"];
    const input = mount(<SelectInput
          options={options}
          onChange={onChange}
          value="area"
          keyName="selectInputPlot"
          />);

    expect(onChange.mock.calls.length).toBe(1);
    input.find({keyName: "selectInputPlot"}).simulate('change', {target: {value: "bar"}})

    expect(onChange.mock.calls.length).toBe(2);
  });
});
