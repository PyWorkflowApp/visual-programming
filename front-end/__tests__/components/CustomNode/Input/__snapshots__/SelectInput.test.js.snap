// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validates SelectInput Display SelectInput 1`] = `ShallowWrapper {}`;
